storage-s3-dynamodb/src/main/java/io/delta/storage/utils/ReflectionUtils.java

```diff
@@ -0,0 +1,37 @@
+package io.delta.storage.utils;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import org.apache.hadoop.conf.Configuration;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+
+public class ReflectionUtils {
+
+
+    private static boolean readsCredsFromHadoopConf(Class<?> awsCredentialsProviderClass) {
+        return Arrays.stream(awsCredentialsProviderClass.getConstructors())
+                .anyMatch(constructor -> constructor.getParameterCount() == 1 &&
+                        Arrays.equals(constructor.getParameterTypes(), new Class[]{Configuration.class}));
+    }
+
+    public static AWSCredentialsProvider createAwsCredentialsProvider(
+            String credentialsProviderName,
+            Configuration initHadoopConf) throws ClassNotFoundException,
```

Comment:
you could just say throws `ReflectiveOperationException` (super class for all of these). Or just capture these and wrap it in a IOException and remove these in the catch block in S3DynamoDBLogStore where this method is used.